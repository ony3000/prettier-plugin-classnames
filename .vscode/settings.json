{
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnSave": true,
  "eslint.workingDirectories": [{ "mode": "auto" }]
}
