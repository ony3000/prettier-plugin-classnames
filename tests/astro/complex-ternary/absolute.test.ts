import { thisPlugin, testSnapshotEach } from '../../adaptor';
import { baseOptions } from '../../settings';
import { fixtures } from './fixtures';

const options = {
  ...baseOptions,
  plugins: ['prettier-plugin-astro', thisPlugin],
  parser: 'astro',
  endingPosition: 'absolute',
};

testSnapshotEach(fixtures, options);
