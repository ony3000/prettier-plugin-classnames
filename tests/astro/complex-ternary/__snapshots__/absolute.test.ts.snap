// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'(1) nested ternary - string literal ternary' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? "lorem ipsum dolor sit amet"
        : condition
          ? \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin\`
          : \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin ex massa hendrerit eu
            posuere\`}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(2) nested ternary - template literal ternary' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? "lorem ipsum dolor sit amet"
        : condition
          ? \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin\`
          : \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin ex massa hendrerit eu
            posuere\`}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(3) nested ternary - nested expression in falsy part' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? "lorem ipsum dolor sit amet"
        : condition
          ? \`lorem ipsum dolor sit amet
            \${"consectetur adipiscing elit proin"} ex massa
            hendrerit eu posuere\`
          : \`lorem ipsum dolor sit amet
            \${"consectetur adipiscing elit proin"} ex massa
            hendrerit eu posuere\`}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(4) nested ternary - nested expression in truthy part' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? condition
          ? \`lorem ipsum dolor sit amet
            \${"consectetur adipiscing elit proin"} ex massa
            hendrerit eu posuere\`
          : \`lorem ipsum dolor sit amet
            \${"consectetur adipiscing elit proin"} ex massa
            hendrerit eu posuere\`
        : "lorem ipsum dolor sit amet"}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(5) double nested ternary - string literal ternary' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? "lorem ipsum dolor sit amet"
        : condition
          ? \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin\`
          : condition
            ? \`lorem ipsum dolor sit amet consectetur
              adipiscing elit proin\`
            : \`lorem ipsum dolor sit amet consectetur
              adipiscing elit proin ex massa hendrerit eu
              posuere\`}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(6) double nested ternary - template literal ternary' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? "lorem ipsum dolor sit amet"
        : condition
          ? \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin\`
          : condition
            ? \`lorem ipsum dolor sit amet consectetur
              adipiscing elit proin\`
            : \`lorem ipsum dolor sit amet consectetur
              adipiscing elit proin ex massa hendrerit eu
              posuere\`}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(7) double nested ternary - nested expression in falsy part' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? "lorem ipsum dolor sit amet"
        : condition
          ? \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin\`
          : condition
            ? \`lorem ipsum dolor sit amet
              \${"consectetur adipiscing elit proin"} ex
              massa hendrerit eu posuere\`
            : \`lorem ipsum dolor sit amet
              \${"consectetur adipiscing elit proin"} ex
              massa hendrerit eu posuere\`}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(8) double nested ternary - nested expression in truthy part' > expectation 1`] = `
"<div>
  <div>
    <div
      class={condition
        ? condition
          ? condition
            ? \`lorem ipsum dolor sit amet
              \${"consectetur adipiscing elit proin"} ex
              massa hendrerit eu posuere\`
            : \`lorem ipsum dolor sit amet
              \${"consectetur adipiscing elit proin"} ex
              massa hendrerit eu posuere\`
          : \`lorem ipsum dolor sit amet consectetur
            adipiscing elit proin\`
        : "lorem ipsum dolor sit amet"}
    >
      <slot />
    </div>
  </div>
</div>
"
`;
